src/components/SplitButton/__stories__/splitButton.stories.mdx

```diff
@@ -0,0 +1,146 @@
+<Canvas>
+  <Story name="Button types">
+    <SplitButton children="Primary" />
+    <SplitButton children="Secondary" kind={SplitButton.kinds.SECONDARY} />
+    <SplitButton children="Tertiary" kind={SplitButton.kinds.TERTIARY} />
+  </Story>
+</Canvas>
+
+### Icons
+<Canvas>
+  <Story name="Icons">
+    <SplitButton leftIcon={Check} children="Left icon" />
+    <SplitButton rightIcon={Check} children="Right icon" />
+    <SplitButton leftIcon={Check} />
+  </Story>
+</Canvas>
+
+## Do’s and Don’ts
+<ComponentRules
+  rules={[
+    {
+      positive: {
+        component: [
+          <SplitButton children="New item" />,
+          <SplitButton success kind={SplitButton.kinds.TERTIARY} leftIcon={Search}>Search</SplitButton>,
+          <Button kind={Button.kinds.TERTIARY} leftIcon={Filter}>Filter</Button>
+        ],
+        description: "Use only one primary action within a single view."
+      },
+      negative: {
+        component: [<SplitButton children="New item" />,<Button>Search</Button>],
+        description: "Don’t use multiple primary buttons within a single view."
+      }
+    },
+    {
+      positive: {
+        component: <SplitButton children="New Item" secondaryDialogContent={() => SplitButtonExampleDialog({  options: [{ text: "New group"}, {text: "New board"}] })}/>,
+        description: "Use split button if there are more than one action within the menu."
+      },
+      negative: {
+        component: <SplitButton children="New Item" secondaryDialogContent={() => SplitButtonExampleDialog({  options: [{ text: "New group"}] })}/>,
+        description: <>Don’t use split button if there’s only one option within the menu. Use <Link href="/?path=/docs/components-button--overview" withoutSpacing>Button</Link> instead.</>
+      }
+    },
+    {
+      positive: {
+        component: <SplitButton children="New Item" secondaryDialogContent={() => SplitButtonExampleDialog({  options: [{ icon: Group, text: "New group"}, { icon: Upload, text: "New board"}] })}/>,
+        description: "Use a split button to display an action with related actions. The main action should be the most common."
+      },
+      negative: {
+        component: <SplitButton children="New Item" secondaryDialogContent={() => SplitButtonExampleDialog({  options: [{ icon: Print, text: "Print board"}, { icon: Sort, text: "Sort items"}] })}/>,
+        description: "Don’t use a split button just to group actions."
+      }
+    }
+  ]}
+/>
+
+## Use cases and examples
+### Split button as the primary action
+Use template is the main action.
+<Canvas>
+  <Story name="Split button as the primary action">
+    <SplitButton 
+      className="monday-storybook-split-button_spacing"
+      children="Use template"
+      secondaryPositions={SplitButton.secondaryPositions.BOTTOM_END}
+      secondaryDialogContent={() => SplitButtonExampleDialog({ options: [{icon: Download, text: "Import template"}, { icon: Upload, text: "Export template"}] }) } 
```

Comment:
please wrap the story with anoynimus functions (like we did in toast) and move the content of the secondaryDialogContent prop to a const variable using useCallback (as we did in the toast)